pwnlib/dynelf.py

```diff
@@ -178,7 +178,7 @@ def __init__(self, leak, pointer=None, elf=None, libcdb=True):
 
         self.elf     = elf
         self.leak    = leak
-        self.libbase = self._find_base(pointer or elf.address)
+        self.libbase = elf.address if elf else self._find_base(pointer)
```

Comment:
Changing this will make incorrect `ELF` addresses break DynELF.  By providing the address to `_find_base`, we can verify that the address actually has the `\x7fELF` header.

Reply:
I agree with you after thinking more about it. My reasoning was that it was silly not to trust the base address of the ELF object, as loads of other stuff breaks, but it doesn't create that many extra lookups and my proposal is hard to override.